src/components/Table/GridTable.stories.tsx

```diff
@@ -1136,3 +1136,99 @@ export function ToggleCustomCollapse() {
     </>
   );
 }
+
+type ExpandHeader = { id: "expandableHeader"; kind: "expandableHeader" };
+type Header = { id: "header"; kind: "header" };
+type ExpandableData = {
```

Comment:
Why doesn't this one need `id`?

Also 98% of places I've seen ID used, with the sole exception of `user:1` below, it's matched -exactly- the `kind`. Why can't we just infer it from `kind` unless they want to specifically override it? `row.id ??= row.kind`.

Reply:
The `id` is a required prop for all `GridRowData` types. We currently use a combination of the `kind` + `id` props to create the unique React `key` for each row.
Potentially for these "reserved" sort of rows, we could infer it, but that'd mean making `row.id` optional in some cases, but not all. This could get tricky... 🤔 
I get your point, seems redundant and unnecessary. But it is a required prop for how everything is organized right now. I don't want to get into a big optional row id refactor at the moment, but could be an interesting exploration in the future. Thanks for calling out!